### PR TITLE
Mark exceptions as stable.  We do not plan to change this behavior in…

### PR DESCRIPTION
**Pull request by jsuereth**
… breaking ways.

We discussed the current exception specification.  Today, the specification defines how exceptions show up in spans.  It's been in use for a while and we don't want to create breaking changes.  We believe we can evolve the specification we have as needed in non-breaking ways.

### CODE REVIEW

**Review by jack-berg — APPROVED** [review]
We've talked about potential long terms plans to instead record span events as event logs records. This isn't a blocker for stabilizing recording exception as span events since anything we do replacing span events with event log records must be done in a way that allows the spec to evolve gracefully.

Stabilizing this is long overdue.

**Review by trask — APPROVED** [review]
I agree. It's hard to argue against stabilizing this since it's part of all of the stable SDKs already.

I'm assuming this means we'll mark the `exception.*` attributes as stable in the semantic-conventions repo?

**Comment by tigrannajaryan** [comment]
I agree with should stabilize. The only question to me is if we are happy with `exception.` as the namespace. This came up in a recent discussion about events. One possible alternate surfaced was `runtime.exception.`

**Comment by jsuereth** [comment]
> I agree with should stabilize. The only question to me is if we are happy with `exception.` as the namespace. This came up in a recent discussion about events. One possible alternate surfaced was `runtime.exception.`

In Semconv we use `reasonably unique` namespaces that are meant to be minimal.  `runtime` carries a lot of baggage here and walks into the `errors` vs. `exceptions` space.  I don't think that discussion is worth revisiting again.

> I'm assuming this means we'll mark the exception.* attributes as stable in the semantic-conventions repo?
Yes, we can/should mark some of these as stable.  This specification update only outlines *what* fields are filled out, not what they contain.

**Review by cijothomas — APPROVED** [review]
Thanks! 

Languages like OTel.NET have shipped this already as part of [stable API](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/src/OpenTelemetry.Api/Trace/ActivityExtensions.cs#L78) un-intentionally for a very long time.!

**Review by vishweshbankwar — APPROVED** [review]

**Review by pyohannes — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -37,6 +37,8 @@ release.
 
 - Add a new AddLink() operation to Span (experimental).
   ([#3678](https://github.com/open-telemetry/opentelemetry-specification/pull/3678))
+- Stabilize SDK exception handling.
```
Comment (arminru) [inline CHANGELOG.md:40]: That sounds as if it was about handling exceptions coming from the SDK, whereas it is about instrumentations recording telemetry data about application exceptions.

```suggestion
- Stabilize how exceptions are recorded using the Trace SDK.
```

Comment (cijothomas) [reply CHANGELOG.md:40]: Stabilize how exceptions are recorded using the Tracing API --> as this is part of API itself.

Comment (jsuereth) [reply CHANGELOG.md:40]: This doesn't affect the API - it only affects how SDKs record exceptions passed in from the API.  I like Armin's wording here.

**Review by arminru — APPROVED** [review]

**Comment by carlosalberto** [comment]
Overall LGTM, but I'm slightly wary of the fact the _actual_ semconv values are mentioned [at the bottom](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/exceptions.md#attributes) of this document. Should we remove them altogether _and_ keep the link to their semconv section?

**Comment by tigrannajaryan** [comment]
@jack-berg I did not attend the last Events WG meeting. Have you had a chance to discuss this?

**Comment by jack-berg** [comment]
> @jack-berg I did not attend the last Events WG meeting. Have you had a chance to discuss this?

No we did not. The 11/24 Event WG is cancelled due to US holiday, so the next opportunity is 12/08. 

Personally, I think there are strong reasons to stabilize this without the agreement of the Event WG, although this is not to say that I anticipate disagreement from the Event WG. I think the Event WG needs to find a way to evolve around this behavior rather than delaying stabilization and shifting course.

**Review by jmacd — APPROVED** [review]

**Comment by lmolkova** [comment]
I assume once it's merged we're going to also mark corresponding attributes as stable in the yaml and [md](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/exceptions/exceptions-spans.md) in the semconv repo?

**Comment by jsuereth** [comment]
@lmolkova See https://github.com/open-telemetry/opentelemetry-specification/pull/3769#issuecomment-1813305044

Yes, we'll be looking to stabilize those too.

**Comment by arminru** [comment]
Not stale, just making sure there was enough time for reviewing before merging it.

**Review by carlosalberto — APPROVED** [review]

**Review by Oberon00 — APPROVED** [review]

**Pull request merged**